auth_oidc/controllers/main.py

```diff
@@ -48,3 +51,28 @@ def list_providers(self):
                     provider["auth_endpoint"], url_encode(params)
                 )
         return providers
+
+
+class OpenIDLogout(Session):
+
+    @http.route("/web/session/logout", type="http", auth="none")
+    def logout(self, redirect="/web/login"):
+        user = request.env["res.users"].sudo().browse(request.session.uid)
+        if user.oauth_provider_id.id:
+            p = request.env["auth.oauth.provider"].sudo().browse(user.oauth_provider_id.id)
+            if p.logout_endpoint:
+                r = redirect
+                if r.find('http') == -1 and r.find('https') == -1:
```

Comment:
I have sneaked in support for full URLs

Reply:
Please use `not r.startswith("http:") and not r.startswith("https:")` here. the check you used would skip adding the base url for all redirects containing http.

Reply:
Done